substrate/frame/revive/src/lib.rs

```diff
@@ -1179,6 +1179,29 @@ where
 	MomentOf<T>: Into<U256>,
 	T::Hash: frame_support::traits::IsType<H256>,
 {
+	/// Ensure the origin has no code deplyoyed if it is a signed origin.
+	fn ensure_non_contract_if_signed(origin: &OriginFor<T>) -> Result<(), DispatchError> {
+		use crate::exec::{code_hash, EMPTY_CODE_HASH};
+		if let Ok(who) = ensure_signed(origin.clone()) {
+			let address = <T::AddressMapper as AddressMapper<T>>::to_address(&who);
+			// Get canonical code-hash (handles precompiles, contract storage and non-existent
+			// accounts).
+			let code_hash = code_hash::<T>(&address);
+
+			// Deployed code exists when hash is neither zero (no account) nor EMPTY_CODE_HASH
+			// (account exists but no code).
+			if code_hash != H256::zero() && code_hash != EMPTY_CODE_HASH {
```

Comment:
Might be a matter of taste but I find this easier to understand.

```suggestion
			if !(code_hash == H256::zero() || code_hash == EMPTY_CODE_HASH) {
```